Tests/SwiftLintFrameworkTests/BaselineTests.swift

```diff
@@ -0,0 +1,222 @@
+        XCTAssertTrue(FileManager.default.changeCurrentDirectoryPath(temporaryDirectoryPath))
+    }
+
+    override final class func tearDown() {
+        if let currentDirectoryPath {
+            XCTAssertTrue(FileManager.default.changeCurrentDirectoryPath(currentDirectoryPath))
+            self.currentDirectoryPath = nil
+        }
+        super.tearDown()
+    }
+
+    func testWritingAndReading() throws {
+        try withExampleFileCreated { sourceFilePath in
+            let baselinePath = temporaryDirectoryPath.stringByAppendingPathComponent(UUID().uuidString)
+            try Baseline(violations: Self.violations(for: sourceFilePath)).write(toPath: baselinePath)
+            defer {
+                try? FileManager.default.removeItem(atPath: baselinePath)
+            }
+            let newBaseline = try Baseline(fromPath: baselinePath)
+            XCTAssertEqual(newBaseline, Self.baseline(for: sourceFilePath))
+        }
+    }
+
+    func testUnchangedViolations() throws {
+        try withExampleFileCreated { sourceFilePath in
+            XCTAssertEqual(Self.baseline(for: sourceFilePath).filter(Self.violations(for: sourceFilePath)), [])
+        }
+    }
+
+    func testShiftedViolations() throws {
+        try withExampleFileCreated { sourceFilePath in
+            let baseline = Self.baseline(for: sourceFilePath)
+            let violations = try Self.violations(for: sourceFilePath).lineShifted(by: 2, path: sourceFilePath)
+            XCTAssertEqual(baseline.filter(violations), [])
+        }
+    }
+
+    func testNewViolation() throws {
+        try testViolationDetection(
+            violationRuleDescriptions: Self.ruleDescriptions,
+            newViolationRuleDescription: EmptyCollectionLiteralRule.description,
+            insertionIndex: 2
+        )
+    }
+
+    func testViolationDetection() throws {
+        let violationRuleDescriptions = [
+            ArrayInitRule.description,
+            BlockBasedKVORule.description,
+            ArrayInitRule.description,
+            ClosingBraceRule.description,
+            ClosingBraceRule.description,
+            ClosingBraceRule.description,
+            BlockBasedKVORule.description,
+            DirectReturnRule.description,
+            ArrayInitRule.description,
+            ClosingBraceRule.description
+        ]
+
+        let ruleDescriptions = [
+            ArrayInitRule.description,
+            BlockBasedKVORule.description,
+            ClosingBraceRule.description,
+            DirectReturnRule.description
+        ]
+
+        for ruleDescription in ruleDescriptions {
+            for insertionIndex in 0..<violationRuleDescriptions.count {
+                try testViolationDetection(
+                    violationRuleDescriptions: violationRuleDescriptions,
+                    newViolationRuleDescription: ruleDescription,
+                    insertionIndex: insertionIndex
+                )
+            }
+            break
```

Comment:
The `break` is unintended here, isn't it?

Reply:
let me check that ...

Reply:
Ah yes 🤦 - it was left over from testing